docs/config/README.md

```diff
@@ -1181,3 +1181,53 @@ The module will be shown if any of the following conditions are met:
 [username]
 disabled = true
 ```
+
+## Custom commands
+
+The `custom` modules show the output of some arbitrary commands.
+
+These modules will be shown if any of the following conditions are met:
+- The current directory contains a file whose name is in `files`
+- The current directory contains a directory whose name is in `directories`
+- The current directory contains a file whose extension is in `extensions`
+- The `when` command returns 0
+
+::: tip
+
+This module can be specified multiple times by using a `.`.
+
+:::
+
+### Options
+
+| Variable      | Default             | Description                                                                 |
+| ------------- | ------------------- | --------------------------------------------------------------------------- |
+| `command`     |                     | The command whose output should be printed.                                 |
+| `when`        |                     | A command that will force the module to be displaed if it returns 0.        |
+| `description` | `"<custom module>"` | The description of the module.                                              |
+| `files`       | `[]`                | The files that will be searched in the working directory for a match.       |
+| `directories` | `[]`                | The directories that will be searched in the working directory for a match. |
+| `extensions`  | `[]`                | The extensions that will be searched in the working directory for a match.  |
+| `symbol`      | `""`                | The symbol used before displaying the command output.                       |
+| `style`       | `"bold green"`      | The style for the module.                                                   |
+| `prefix`      | `""`                | Prefix to display immediately before the command output.                    |
+| `suffix`      | `""`                | Suffix to display immediately after the command output.                     |
+| `disabled`    | `false`             | Disables this `custom` module.                                              |
+
+### Example
+
+```toml
+# ~/.config/starship.toml
+
+prompt_order = [
```

Comment:
I wasn't certain about this either, but since you're okay with that, great! I'm enabling `toml`'s "preserve_order" feature for that, so that we know the order at which the custom modules were defined.

Reply:
There are several ways to do this. Should I add an additional (magic-ish) "custom" module that adds all custom modules in the order they were defined and add it to the `prompt_order` (and cie), or should I handle that more directly in the printer?

Reply:
I think the former sounds like a clearer solution if it's not too difficult to do.
It would possibly allow users to define the location of "custom" in `prompt_order` as a group, if they choose to do so.

Reply:
Here's what I did:
- Create a special "custom" module
  * It is special because it actually returns a vector of modules, rather than a list of modules.
  * It is therefore handled directly in `print.rs`.
- Add it to the default prompt order before `line_break`.
- When it gets loaded, it adds all custom modules **that are not explicitly added in `prompt_order`** in the order they were defined in the modules to print.